services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -0,0 +1,858 @@
+        def flowInfoFromDb = database.getFlow(flow.flowId)
+
+        and: "Update switch properties(multi_table: true) on the transit switch"
+        northbound.updateSwitchProperties(involvedSwitches[1].dpId,
+                northbound.getSwitchProperties(involvedSwitches[1].dpId).tap { it.multiTable = true })
+
+        and: "Fail a flow ISL (bring switch port down)"
+        def flowIsls = pathHelper.getInvolvedIsls(desiredPath)
+        def islToBreak = flowIsls[0]
+        antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        //flow is pinned, so that's why the flow is not rerouted
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow.flowId).status == FlowState.DOWN
+            assert pathHelper.convert(northbound.getFlowPath(flow.flowId)) == desiredPath
+        }
+
+        and: "Update switch properties(multi_table: true) on the dst switch"
+        northbound.updateSwitchProperties(involvedSwitches[2].dpId,
+                northbound.getSwitchProperties(involvedSwitches[2].dpId).tap { it.multiTable = true })
+
+        and: "Restore the failed flow ISL (bring switch port up)"
+        antiflap.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP
+            assert pathHelper.convert(northbound.getFlowPath(flow.flowId)) == desiredPath
+        }
+
+        then: "Flow rules are still in single table mode on all transit switches"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            verifyAll(northbound.getSwitchRules(involvedSwitches[0].dpId).flowEntries) { rules ->
+                rules.find { it.cookie == flowInfoFromDb.forwardPath.cookie.value }.tableId == 0
+                rules.find { it.cookie == flowInfoFromDb.reversePath.cookie.value }.tableId == 0
+            }
+            verifyAll(northbound.getSwitchRules(involvedSwitches[1].dpId).flowEntries) { rules ->
+                rules.find { it.cookie == flowInfoFromDb.forwardPath.cookie.value }.tableId == 0
+                rules.find { it.cookie == flowInfoFromDb.reversePath.cookie.value }.tableId == 0
+            }
+            verifyAll(northbound.getSwitchRules(involvedSwitches[2].dpId).flowEntries) { rules ->
+                rules.find { it.cookie == flowInfoFromDb.forwardPath.cookie.value }.tableId == 0
+                rules.find { it.cookie == flowInfoFromDb.reversePath.cookie.value }.tableId == 0
+            }
+        }
+
+        and: "Flow is valid and pingable"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        when: "Synchronize the flow"
+        with(northbound.synchronizeFlow(flow.flowId)) { it.rerouted }
```

Comment:
This operation is SO not interesting. You already did it million times before. Why you never try to synchronize switch for example? This is a far more common operation

Reply:
in that test everything is the other way around.
here the flow is rerouted.